pkg/apis/migration/v1alpha1/migplan_types.go

```diff
@@ -144,3 +145,89 @@ func (r *MigPlan) GetRefResources(client k8sclient.Client) (*PlanResources, erro
 
 	return resources, nil
 }
+
+// PV Actions.
+const (
+	PvMoveAction = "move"
+	PvCopyAction = "copy"
+)
+
+// Name - The PV name.
+// StorageClass - The PV storage class name.
+// Action - The PV migration action (move|copy)
+// staged - A PV has been explicitly added/updated.
+type PersistentVolume struct {
+	Name         string `json:"name,omitempty"`
+	StorageClass string `json:"storageClass,omitempty"`
+	Action       string `json:"action,omitempty"`
+	staged       bool
+}
+
+// Collection of PVs
+type PersistentVolumes struct {
+	Content []PersistentVolume `json:"persistentVolumes"`
```

Comment:
Any thoughts on making this a map? Since everytime you are adding a new PV you have to loop through content to see if it already exists. 

Then `FindPv` becomes:
```
return r.Content[pv.Name]
```

Probably not a huge deal but if we are doing a migration with 100 PVs this could affect performance

Reply:
It's a `slice` primarily because of how it will serialize in the API.  Agreed that a map would be more efficient.  I'll play with it and see what I can do.

Reply:
Actually, I have an idea about how to do this easily.  I'll play with it.

Reply:
np!

Reply:
@dymurray I added an _index_ which seemed less complex than a custom field tag that would be needed to json serialize a `map` as an array.

Reply:
+1 Looks great!